MapboxNavigation/NavigationViewControllerDelegate.swift

```diff
@@ -185,18 +185,123 @@ public protocol NavigationViewControllerDelegate: VisualInstructionDelegate {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, simplifiedShapeFor route: Route) -> MGLShape? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, waypointStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, waypointSymbolStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, shapeFor waypoints: [Waypoint], legIndex: Int) -> MGLShape? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, didSelect route: Route) {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, mapViewUserAnchorPoint mapView: NavigationMapView) -> CGPoint {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .info)
+        return .zero
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, shouldDiscard location: CLLocation) -> Bool {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return RouteController.DefaultBehavior.shouldDiscardLocation
+        
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, roadNameAt location: CLLocation) -> String? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, imageFor annotation: MGLAnnotation) -> MGLAnnotationImage? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
+        return nil
+    }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, viewFor annotation: MGLAnnotation) -> MGLAnnotationView? {
+        logUnimplemented(protocolType: NavigationViewControllerDelegate.self,  level: .debug)
```

Comment:
How can we automatically verify that all the methods of an UnimplementedLogging-conforming class call `logUnimplemented(protocolType:level:)`? In other languages, this is what you’d use a custom decorator or annotation for. Maybe an automated test could use SourceKitten to compare the number of methods and calls to `logUnimplemented(protocolType:level:)`? Or perhaps UnimplementedLogging could require the class to register each logged method into an array, and the protocol could check at runtime whether any methods are missing from that array.